variable/test/variable_test.cpp

```diff
@@ -841,17 +841,8 @@ TEST(Variable, self_nesting_scalar_copy) {
 TEST(Variable, self_nesting_scalar_move) {
   Variable inner = makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{2, 3});
 
-  // 1 level of nesting
-  Variable v1 = makeVariable<Variable>(Shape{}, Values{copy(inner)});
-  ASSERT_NO_THROW_DISCARD(v1 = std::move(v1));
-  ASSERT_EQ(v1.value<Variable>(), inner);
-  v1 = makeVariable<Variable>(Shape{}, Values{inner});
-  ASSERT_THROW_DISCARD(v1.value<Variable>() = std::move(v1),
-                       std::invalid_argument);
-  ASSERT_EQ(v1.value<Variable>(), inner);
-  v1 = makeVariable<Variable>(Shape{}, Values{inner});
```

Comment:
I see why the line above this was removed, but what about these?

Reply:
Right, I removed too much

Reply:
Actually, no, `v1.value<Variable>() = std::move(v1)` is also pretty strange.

Reply:
I think these tests are problably from a time when we still had unique_ptr and are much less relevant with shared_ptr, since we rarely (if ever) move variables.